packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts

```diff
@@ -108,6 +108,15 @@ export class NodeModuleCopyHelper extends FileCopyHelper {
               return null
             }
 
+            if (stat.isSymbolicLink()) {
+              const resolvedLinkTarget = realpathSync(filePath)
+              if (!isSubPath(resolvedLinkTarget, tmpPath)) {
+                // delete symlink file if it links outside module
+                log.warn({ module: moduleName, file: filePath, resolvedLinkTarget }, `deleting symlink outside module`)
+                return null
+              }
+            }
```

Comment:
I think this logic needs to be in `asarUtil` instead of `NodeModuleCopyHelper` as the symlink outside the project dir can't be allowed for both file assets or node modules.

Let me finish up the `electron/asar` migration and I can incorporate your changes here into that PR branch

Reply:
If it's done in asarUtil, any files linked to other workspaces within the project will all fail. I'm not sure how common this usage pattern is.

Reply:
>any files linked to other workspaces within the project will all fail

Can you elaborate on this? Are you referring to a two-package.json project structure?

>I'm not sure how common this usage pattern is.

It's less so about how common it is and more so about enforcing best practices. For instance, electron/asar already validates making sure symlinks are within the package during asar extraction: https://github.com/electron/asar/blob/464e83436967438c74d8ac184b088fc780706b2d/src/asar.ts#L249-L253

Reply:
Agreeded. I will close it.